File: computingservices/DedupeServices/utils/foiredisstreamdb.py
from walrus import Database
import os

from dotenv import load_dotenv
load_dotenv()

redishost = os.getenv('REDIS_HOST') 
redisport = os.getenv('REDIS_PORT')
redispassword = os.getenv('REDIS_PASSWORD')


redisstreamdb =  Database(host=str(redishost), port=str(redisport), db=0,password=str(redispassword))

File: computingservices/OCRServices/utils/foi_redisstream_db.go
package utils

import (
	"context"
	"fmt"
	"os"

	"github.com/go-redis/redis/v8"
)

var (
	ctx = context.Background()
)

func CreateRedisClient() *redis.Client {
	host := os.Getenv("REDIS_HOST")
	port := os.Getenv("REDIS_PORT")
	password := os.Getenv("REDIS_PASSWORD")
	addr := fmt.Sprintf("%s:%s", host, port)
	rdb := redis.NewClient(&redis.Options{
		Addr:     addr,
		Password: password,
		DB:       0,
	})
	_, err := rdb.Ping(ctx).Result()
	if err != nil {
		panic(fmt.Sprintf("Failed to connect to Redis: %v", err))
	}
	return rdb
}
